perses/app/relative_point_mutation_setup.py

```diff
@@ -215,18 +220,22 @@ def __init__(self,
                 _logger.warning(f'ligand filetype not recognised. Please provide a path to a .pdb or .sdf file')
                 return
 
-            # Now create a complex
-            complex_md_topology = protein_md_topology.join(ligand_md_topology)
-            complex_topology = complex_md_topology.to_openmm()
-            complex_positions = unit.Quantity(np.zeros([protein_n_atoms + ligand_n_atoms, 3]), unit=unit.nanometers)
-            complex_positions[:protein_n_atoms, :] = protein_positions
-            complex_positions[protein_n_atoms:, :] = ligand_positions
-
-            # Convert positions back to openmm vec3 objects
-            complex_positions_vec3 = []
-            for position in complex_positions:
-                complex_positions_vec3.append(openmm.Vec3(*position.value_in_unit_system(unit.md_unit_system)))
-            complex_positions = unit.Quantity(value=complex_positions_vec3, unit=unit.nanometer)
+            if solvate:
+                # Now create a complex
+                complex_md_topology = protein_md_topology.join(ligand_md_topology)
+                complex_topology = complex_md_topology.to_openmm()
+                complex_positions = unit.Quantity(np.zeros([protein_n_atoms + ligand_n_atoms, 3]), unit=unit.nanometers)
+                complex_positions[:protein_n_atoms, :] = protein_positions
+                complex_positions[protein_n_atoms:, :] = ligand_positions
+
+                # Convert positions back to openmm vec3 objects
+                complex_positions_vec3 = []
+                for position in complex_positions:
+                    complex_positions_vec3.append(openmm.Vec3(*position.value_in_unit_system(unit.md_unit_system)))
+                complex_positions = unit.Quantity(value=complex_positions_vec3, unit=unit.nanometer)
+            else:
+                complex_topology = ligand_topology
+                complex_positions = ligand_positions
```

Comment:
the exception for pre-existing waters could go here i guess?

Reply:
yes, that's the whole point of the `else` part of this logical branch. if the topology is already solvated, we're assuming its the full complex (not just the ligand), so we don't need to combine topologies anymore.